src/step-function.ts

```diff
@@ -466,6 +468,114 @@ export namespace $SFN {
       });
     },
   });
+
+  type AWSServiceClass = { new (): AWSService };
+
+  /**
+   * First we have to extract the names of all Services in the v2 AWS namespace
+   *
+   * @returns "AccessAnalyzer" | "Account" | ... | "XRay"
+   */
+  type ServiceKeys = {
+    [K in keyof typeof AWS]: typeof AWS[K] extends AWSServiceClass ? K : never;
+  }[keyof typeof AWS];
+
+  /**
+   * We define this to omit any non-funciton types from service
+   * instances. e.g. `apiVersions` and `config`
+   */
+  type ServiceInstance<T extends AWSServiceClass> = {
+    [K in FunctionKeys<InstanceType<T>>]: InstanceType<T>[K];
+  };
+
+  export const SDK: {
+    [serviceName in ServiceKeys]: ServiceInstance<typeof AWS[serviceName]>;
+  } = new Proxy({} as any, {
```

Comment:
We should move to this beauty https://github.com/functionless/functionless/issues/405#issuecomment-1213036271